fix: Magnetic modifier doesn't fully work on big element

juleshry opened the pull request:
resolves issue [#18](https://github.com/holux-design/v-gsap-nuxt/issues/18)

I spotted multiple problems.
- the magneticDistance depended on the strengthFactor 
- the x and y axes were handled the same so only square-shaped elements would fit
- the computed distance between the mouse and the element is based on the center of the element and doesn't depend on its size 

Here's what I implemented to fix those issues
- 2 distances : distance to the element and distance to its center
- magneticDistance only depends on the element size 
- x and y axes are handled separately for the magneticDistance resulting in a more accurate magneticDistance for non square-shaped elements.

What do you think ? Does this fit what you had in mind ?

holux-design commented:
Thats amazing! Thank you!

However I feel like this needs some more fine-tuning. I tried comparing current with your branch and it seems that a) it got quite a bit stronger (which should easily be fixable) but also b) on approach the element "jumps" towards the mouse, while on moving-away the element is dragged for quite a distance before relase.

I tried making a GIF of it, left is original, right is PR:
![CleanShot 2025-03-06 at 15 19 06](https://github.com/user-attachments/assets/8c53242d-4780-40f0-bf51-2cac46f85009)

You think that can be adapted to be "closer" to the original behaviour while preserving your obvious improvements?


Cheers

juleshry commented:
@holux-design I updated the magnetic strength to behave more like the previous version. Tell me if it's good enough please.

I also added an intersectionObserver to avoid useless calculations.

holux-design commented:
@juleshry Feels good now. Only thing thats still broken is `.refuse``
![CleanShot 2025-03-11 at 20 37 51](https://github.com/user-attachments/assets/eb7dd86f-43c6-4322-8afb-a9f4a0e92a3e)

However, I am willing to cut it off - it was a joke anyways. Can you remove `.refuse` from all occurences in the plugin and in the docs in your PR?

juleshry commented:
@holux-design done ! :)
